### PR TITLE
docs: add design document about auto gc safepoint

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

Publish document about auto gc safepoint. Complete item-7 in https://github.com/pingcap/ticdc/issues/722

Original from:
[自适应 GC safepoint 设计 internal only](https://docs.google.com/document/d/1IWMSXWe-7k111h9yCfrvbQO9TE3c604-ktC9FY1Yb-M/edit?ts=5e8c9dc9#heading=h.c147q4rduzfh)
[自适应 GC safepoint 设计与实现 internal only](https://docs.google.com/document/d/1oMwy5tKR8g-lCK8qlya1TpTvRQH-TVVBvF3brPXDHGA/edit#)

### What is changed and how it works?

Add document

### Check List 

Tests 

 - No code

### Release note

- No release note

### CODE REVIEW

**Comment by overvenus** [comment]
@liuzix PTAL, thanks!

File: docs/design/2020-04-13-ticdc-auto-gc-safepoint-2-cn.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+    ...
+
+    serviceID := string(request.ServiceId)
+    safePoint := request.SafePoint
+    ttl := request.TTL
+    minSafePoint := atomic.LoadUint64(&s.minSafePoint)
+
+    // Only save the safe point if it's greater than the previous one
+    if safePoint > minSafePoint {
```
Comment (YuJuncen) [inline docs/design/2020-04-13-ticdc-auto-gc-safepoint-2-cn.md:96]:
```suggestion
    if safePoint >= minSafePoint {
```

see https://github.com/pingcap/pd/pull/2633. Maybe keep it consistency with the current version would be better?

**Comment by amyangfei** [comment]
PTAL @YuJuncen thanks!

**Pull request merged**